### PR TITLE
EntityTrigger Logic apps for Restrict IP, Domain, Url and FileHash

### PR DESCRIPTION
**Pull request by FathomlessFunction**
Required items, please complete
   
   Change(s):
   - Added Entity Triggered versions of:
      - Restrict MDE FIleHash
      - Restrict MDE IP Address
      - Restrict MDE URL
      - Restrict MDE Domain 

   Reason for Change(s):
   - Current versions of Logic App will loop through and restrict all entities in the incident (for example, Restrict-MDEIPAddress will restrict all IP addresses in an incident), whereas the entity triggered version will allow responders to run the logic app on specific entities in the incident. Logic is the same, aside from the trigger change, and removal of the loop.

   Testing Completed:
   - Yes

### CODE REVIEW

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/Restrict-MDEFileHash-entity-trigger/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,226 @@
+{
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/Restrict-MDEFileHash-entity-trigger/azuredeploy.json:1]: Similar to above, handle the scenario when ARMId doesn't exist.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md [hunk 2]
```diff
@@ -13,13 +13,23 @@ After deployment, attach this playbook to an **automation rule** so it runs when
...
+After deployment, you can run this playbook manually on an alert or attach it to an **analytics rule** so it will run when an alert is created.
 
 [![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEFileHash%2FRestrict-MDEFileHash-alert-trigger%2Fazuredeploy.json)
 [![Deploy to Azure Gov](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEFileHash%2FRestrict-MDEFileHash-alert-trigger%2Fazuredeploy.json)
 
+
+**Deploy with entity trigger**
+
+After deployment, you can run this playbook manually on any incident with a File Hash entity. This will also let you restrict specific file hashes, rather than restricting all file hashes in an incident, as the other triggers will doß.
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md:27]: Correct the spelling 'doß'.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEDomain/readme.md [hunk 3]
```diff
@@ -28,6 +28,17 @@ After deployment, you can run this playbook manually on an alert or attach it to
...
+
+After deployment, you can run this playbook manually on any Sentinel incident with a Domain entity in it. This will also allow you to restrict only specific domains within an incident, rather than restricting all domains associated with an incident.
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEDomain%2FRestrict-MDEDomain-entity-trigger%2Fazuredeploy.json" target="_blank">
+    <img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a>
+<a href="https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEDomain%2FRestrict-MDEDomain-entity-trigger%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.png"/>
+</a>
+
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEDomain/readme.md:41]: In readme file Image links are broken and entity trigger screenshot is missing. 
image

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md [hunk 4]
```diff
@@ -41,4 +51,7 @@ New-AzureAdServiceAppRoleAssignment -ObjectId $MI.ObjectId -PrincipalId $MI.Obje
 ![Incident Trigger](./Restrict-MDEFileHash/Restrict-MDEFileHash-incident-trigger/images/designerLight.png)
 
 **Alert Trigger**
-![Alert Trigger](./Restrict-MDEFileHash/Restrict-MDEFileHash-alert-trigger/images/Restrict-MDEFileHash_alert.png)
\ No newline at end of file
+![Alert Trigger](./Restrict-MDEFileHash/Restrict-MDEFileHash-alert-trigger/images/Restrict-MDEFileHash_alert.png)
+
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md:55]: Image links are broken.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/Restrict-MDEIPAddress-entity-trigger/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
+{
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/Restrict-MDEIPAddress-entity-trigger/azuredeploy.json:1]: Handle the scenario when ARMId doesn't exist.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/readme.md [hunk 6]
```diff
@@ -47,4 +57,7 @@ New-AzRoleAssignment -ObjectId $MIGuid -RoleDefinitionName $RoleName -Scope /sub
 ![Incident Trigger](./Restrict-MDEIPAddress/Restrict-MDEIPAddress-incident-trigger/images/designerLight.png)
 
 **Alert Trigger**
-![Alert Trigger](./Restrict-MDEIPAddress/Restrict-MDEIPAddress-alert-trigger/images/Restrict-MDEIpAddress_alert.png)
\ No newline at end of file
+![Alert Trigger](./Restrict-MDEIPAddress/Restrict-MDEIPAddress-alert-trigger/images/Restrict-MDEIpAddress_alert.png)
+
+**Entity Trigger**
+![Entity Trigger](./Restrict-MDEIPAddress-entity-trigger/images/Restrict-MDEIPAddress-entityTrigger.png)
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/readme.md:63]: Image links are broken.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/readme.md [hunk 7]
```diff
@@ -13,16 +13,26 @@ After deployment, attach this playbook to an **automation rule** so it runs when
 [![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEIPAddress%2FRestrict-MDEIPAddress-incident-trigger%2Fazuredeploy.json)
 [![Deploy to Azure Gov](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FMicrosoftDefenderForEndpoint%2FPlaybooks%2FRestrict-MDEIPAddress%2FRestrict-MDEIPAddress-incident-trigger%2Fazuredeploy.json)
 
+
 **Deploy with alert trigger**
 
 After deployment, you can run this playbook manually on an alert or attach it to an **analytics rule** so it will rune when an alert is created.
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEIPAddress/readme.md:19]: Correct the spelling 'rune'.

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEUrl/Restrict-MDEUrl-entity-trigger/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+"displayName": "[variables('MicrosoftSentinelConnectionName')]",
+"customParameterValues": {},
+"parameterValueType": "Alternative",
+"api": {
+"id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location, '/managedApis/Azuresentinel')]"
+}
+}
+}
+]
+}
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEUrl/Restrict-MDEUrl-entity-trigger/azuredeploy.json:154]: 1. Can you correct the code indentation?
2. Handle the scenariHandle the scenario when ARMId doesn't exist?

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEUrl/readme.md [hunk 9]
```diff
@@ -18,6 +18,12 @@ After deployment, you can run this playbook manually on an alert or attach it to
```
Comment (rahul0216) [inline Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEUrl/readme.md:18]: 1. Correct the spelling 'rune'.
2. In readme file Image links are broken and entity trigger screenshot is missing.

**Review by rahul0216 — COMMENTED** [review]
1. Please address the inline comments.
2. Attach the playbook screenshots in dark mode as well.

**Comment by FathomlessFunction** [comment]
> 1. Please address the inline comments.
> 2. Attach the playbook screenshots in dark mode as well.

Thank you very much for your review! I believe these issues have been resolved in my latest 2 commits, which added:
- dark mode screenshots
- fixes to the image paths
- conditions to check that ARM ID is defined before adding a comment
- correct JSON indentation
- typo fixes

Please let me know if anything else should be addressed, very happy to fix whatever is needed.

**Review by rahul0216 — APPROVED** [review]
Changes look good.

**Review by rahul0216 — CHANGES_REQUESTED** [review]
There is one small change pending in file
Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md

File: Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md (hunk 2 above)
Comment (FathomlessFunction) [reply Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md]: I believe this has been resolved in one of my subsequent Pull Requests.

**Comment by FathomlessFunction** [comment]
> There is one small change pending in file Solutions/MicrosoftDefenderForEndpoint/Playbooks/Restrict-MDEFileHash/readme.md

Thank you very much! Would you mind highlighting which change needs to be made? I believe I have addressed all of your prior comments.

**Review by rahul0216 — APPROVED** [review]
Changes look good. Thanks

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**